src/NosCore.GameObject/Services/CharacterBuilder/CharacterBuilderService.cs

```diff
@@ -19,23 +19,27 @@
 
 using Mapster;
 using NosCore.Data.AliveEntities;
+using NosCore.GameObject.Services.ExchangeInfo;
 using NosCore.GameObject.Services.Inventory;
 
 namespace NosCore.GameObject.Services.CharacterBuilder
 {
     public class CharacterBuilderService : ICharacterBuilderService
     {
         private readonly IInventoryService _inventory;
+        private readonly IExchangeService _exchangeInfo;
 
-        public CharacterBuilderService(IInventoryService inventory)
+        public CharacterBuilderService(IInventoryService inventory, IExchangeService exchangeInfo)
         {
             _inventory = inventory;
+            _exchangeInfo = exchangeInfo;
```

Comment:
Should be build a lot before (for all character to share the same exchange service and should be rename exchangeService